### PR TITLE
Use reference priority extension

### PR DESCRIPTION
**Pull request by rcourtier**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
Fixes https://github.com/powsybl/powsybl-core/issues/2900

**What kind of change does this PR introduce?**
Fixes angle bus reference.

**What is the current behavior?**
Angle bus reference was modelled through the use of SlackTerminal.

**What is the new behavior (if this is a feature change)?**
Angle bus reference is now modelled through the use of ReferencePriorities/ReferencePriority.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java [hunk 1]
```diff
@@ -271,8 +272,8 @@ private static void writeSynchronousMachines(Network network, String cimNamespac
             writer.writeCharacters(CgmesExportUtil.format(-g.getTargetQ()));
             writer.writeEndElement();
             writer.writeStartElement(cimNamespace, "SynchronousMachine.referencePriority");
-            // reference priority is used for angle reference selection (slack)
-            writer.writeCharacters(isInSlackBus(g) ? "1" : "0");
+            // reference priority is used for angle reference selection
+            writer.writeCharacters(Integer.toString(ReferencePriority.get(g)));
```
Comment (jeandemanged) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java:276]: although I don't think CGMES has batteries modelling... same change to do also for batteries further below, I'd expect the `isInSlackBus` method not to be used anywhere also

**Review by jeandemanged — CHANGES_REQUESTED** [review]
note that OLF will write reference terminals only with referenceBusSelectionMode set to GENERATOR_REFERENCE_PRIORITY.

we might want to improve OLF to have it writing reference terminals in FIRST_SLACK mode, by selecting whatever terminal connected to the reference bus.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java (hunk 1 above)
Comment (rcourtier) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/SteadyStateHypothesisExport.java:276]: done

**Review by jeandemanged — APPROVED** [review]

**Comment by annetill** [comment]
https://github.com/powsybl/powsybl-open-loadflow/pull/990

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/export/StateVariablesExportTest.java [hunk 2]
```diff
@@ -391,6 +397,35 @@ void testTopologicalIslandSolvedBusBranch() {
...
+        Terminal t1 = g1.getTerminal();
+        assertTrue(t1.disconnect());
+        assertFalse(t1.isConnected());
+        assertNull(t1.getBusView().getBus());
+
+        // Run a load flow and verify it converged
+        new LoadFlowResultsCompletion().run(network, null);
+        Path outputPath = fileSystem.getPath("tmp-referenceTerminal");
+        Path outputSv = fileSystem.getPath("tmp-referenceTerminal_SV.xml");
+        network.write("CGMES", new Properties(), outputPath);
```
Comment (annetill) [inline cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/export/StateVariablesExportTest.java:425]: The final test you have to do is to check that the angel reference is in the output file.

Comment (rcourtier) [reply cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/export/StateVariablesExportTest.java:425]: done

**Review by annetill — APPROVED** [review]

**Pull request merged**